src/main/webapp/app/plagiarism/manage/exercise-update-plagiarism/exercise-update-plagiarism.component.ts

```diff
@@ -1,89 +1,99 @@
 
-    ngOnDestroy() {
-        this.fieldCPCEnabledSubscription?.unsubscribe();
-        this.fieldTreshholdSubscription?.unsubscribe();
-        this.fieldMinScoreSubscription?.unsubscribe();
-        this.fieldMinSizeSubscription?.unsubscribe();
-        this.fieldResponsePeriodSubscription?.unsubscribe();
-    }
+        this.isFormValid = computed(() => this.formStatus() === 'VALID');
+
+        this.form.get('continuousPlagiarismControlEnabled')!.valueChanges.subscribe((enabled: boolean) => {
+            const fields = [
+                'continuousPlagiarismControlPostDueDateChecksEnabled',
+                'similarityThreshold',
+                'minimumScore',
+                'minimumSize',
+                'continuousPlagiarismControlPlagiarismCaseStudentResponsePeriod',
+            ];
+            fields.forEach((key) => {
+                const ctrl = this.form.get(key);
+                if (enabled) {
+                    ctrl?.enable();
+                } else {
+                    ctrl?.disable();
+                }
+            });
+        });
 
-    calculateFormValid(): void {
-        this.formValid = Boolean(
-            !this.exercise.plagiarismDetectionConfig?.continuousPlagiarismControlEnabled ||
-                (this.fieldThreshhold?.valid && this.fieldMinScore?.valid && this.fieldMinSize?.valid && this.fieldResponsePeriod?.valid),
-        );
-        this.formValidChanges.next(this.formValid);
+        effect(() => {
+            this.formValid = this.isFormValid();
+            this.formValidChanges.next(this.formValid);
+        });
     }
 
-    toggleCPCEnabled() {
-        const config = this.exercise.plagiarismDetectionConfig!;
-        const newValue = !config.continuousPlagiarismControlEnabled;
-        config.continuousPlagiarismControlEnabled = newValue;
-        config.continuousPlagiarismControlPostDueDateChecksEnabled = newValue;
+    ngOnInit(): void {
+        const ex = this.exercise();
+        if (ex && !ex.plagiarismDetectionConfig) {
+            ex.plagiarismDetectionConfig = DEFAULT_PLAGIARISM_DETECTION_CONFIG;
+        }
+        this.minimumSizeTooltip = this.getMinimumSizeTooltip();
+
+        this.form.patchValue({
+            continuousPlagiarismControlEnabled: ex?.plagiarismDetectionConfig?.continuousPlagiarismControlEnabled ?? false,
+            continuousPlagiarismControlPostDueDateChecksEnabled: ex?.plagiarismDetectionConfig?.continuousPlagiarismControlPostDueDateChecksEnabled ?? false,
+            similarityThreshold: ex?.plagiarismDetectionConfig?.similarityThreshold ?? null,
+            minimumScore: ex?.plagiarismDetectionConfig?.minimumScore ?? null,
+            minimumSize: ex?.plagiarismDetectionConfig?.minimumSize ?? null,
+            continuousPlagiarismControlPlagiarismCaseStudentResponsePeriod: ex?.plagiarismDetectionConfig?.continuousPlagiarismControlPlagiarismCaseStudentResponsePeriod ?? null,
+        });
     }
```

Comment:
I do not think that it is good to wrap `this.exercise()` in an `ex` variable - from my perspective this makes the code harder to read and understand, what is the issue with using `this.exercise()` on each occurrence?